test/unit/lodash.spec.js

```diff
@@ -0,0 +1,275 @@
+'use strict';
+
+var chai = require('chai');
+var expect = chai.expect;
+var chance = require('chance');
+var _ = require('lodash');
+var datasets = require('../../app/datasets');
+
+describe('Lodash Collection training', function () {
+    describe('countBy', function () {
+        it('should count repeating numbers', function () {
+            expect(_.countBy.apply(_, datasets.countBy())).to.eql({1: 1, 2: 2, 6: 2, 7: 3});
+        });
+        it('should count length of given strings', function () {
+            expect(_.countBy.apply(_, datasets.countBy2())).to.eql({3: 2, 4: 3, 5: 1});
+        });
+    });
+
+    describe('every', function () {
+        it('should return true if all elements pass the predicate "ALIVE" check', function () {
+            var params = datasets.every();
+            var array = params[0];
+            expect(array.length).to.eql(2);
+            array.forEach(function (n) {
+                expect(_.has(n, 'ALIVE')).to.eql(true);
+            });
+            expect(_.every.apply(_, datasets.every())).to.eql(true);
+        });
+        it('should return true if all elements are Booleanly true', function () {
+            var params = datasets.every2();
+            var array = params[0];
+            expect(array.length).to.eql(4);
+            expect(_.every.apply(_, datasets.every2())).to.eql(true);
+        });
+    });
+
+    describe('filter', function () {
+        it('should return filtered array with 2 objects with matching predicate of age', function () {
+            var params = datasets.filter();
+            var array = params[0];
+            expect(array.length).to.eql(5);
+            expect(_.filter.apply(_, datasets.filter())).to.eql([{name: 'Ali', age: 33, active: false},
+                {name: 'Kali', age: 21, active: true}]);
+        });
+        it('should return filtered array with 3 objects and match for age and active', function () {
+            var params = datasets.filter2();
+            var array = params[0];
+            expect(array.length).to.eql(5);
+            expect(_.filter.apply(_, datasets.filter2())).to.eql([{name: 'Bow', age: 23, active: true},
+                {name: 'Row', age: 23, active: true},
+                {name: 'Sou', age: 23, active: true}]);
+        });
+
+    });
+
+    describe('find', function () {
```

Comment:
find: To solve this issue I was forced to look up into _.fill function since I didn't knew it before. Tests indicate that main idea was to pass method(function probably?) - I guess it is ok to leave it that way, but we should change variable name to predicate. Again we are missing tests for other types, especially function predicate which should be explained more clearly.